src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -100,11 +100,16 @@ export function DiscoverSidebar({
   const [fields, setFields] = useState<IndexPatternField[] | null>(null);
   const [fieldFilterState, setFieldFilterState] = useState(getDefaultFieldFilter());
   const services = useMemo(() => getServices(), []);
+  const uiSettings = services.uiSettings;
 
   useEffect(() => {
-    const newFields = getIndexPatternFieldList(selectedIndexPattern, fieldCounts);
+    const newFields = getIndexPatternFieldList(
+      uiSettings.get(MODIFY_COLUMNS_ON_SWITCH),
+      selectedIndexPattern,
+      fieldCounts
+    );
     setFields(newFields);
-  }, [selectedIndexPattern, fieldCounts, hits, services]);
+  }, [selectedIndexPattern, fieldCounts, hits, services, uiSettings]);
```

Comment:
I guess we should add `uiSettings.get(MODIFY_COLUMNS_ON_SWITCH)` into dependency instead of uiSettings.

Reply:
okk!